contracts/CozyInvestConvex.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/ICozyInvest.sol";
+import "./utils/CozyInvestHelpers.sol";
+
+interface IConvexBooster {
+  // Returns: lptoken, token, gauge, crvRewards, stash, shutdown
+  function poolInfo(uint256)
+    external
+    view
+    returns (
+      address,
+      address,
+      address,
+      address,
+      address,
+      bool
+    );
+
+  function depositAll(uint256 _pid, bool _stake) external returns (bool);
+}
+
+interface IConvexRewardManager {
+  function extraRewards(uint256 index) external returns (address);
+
+  function extraRewardsLength() external returns (uint256);
+
+  function getReward(address _account, bool _claimExtras) external returns (bool);
+
+  function rewardToken() external returns (IERC20);
+
+  function withdrawAndUnwrap(uint256 amount, bool claim) external returns (bool);
+}
+
+interface ICrvDepositZap {
+  function add_liquidity(uint256[4] calldata amounts, uint256 minMintAmount) external returns (uint256);
+
+  function add_liquidity(
+    address pool,
+    uint256[4] calldata amounts,
+    uint256 minMintAmount
+  ) external returns (uint256);
+
+  function calc_withdraw_one_coin(
+    address pool,
+    uint256 amount,
+    uint128 index
+  ) external returns (uint256);
+
+  function remove_liquidity_one_coin(
+    uint256 amount,
+    int128 index,
+    uint256 minAmount
+  ) external returns (uint256);
+
+  function remove_liquidity_one_coin(
+    address pool,
+    uint256 amount,
+    int128 index,
+    uint256 minAmount
+  ) external returns (uint256);
+}
+
+/**
+ * @notice On-chain scripts for borrowing from Cozy and using the borrowed funds to
+ * supply to curve and then deposit into Convex
+ * @dev This contract is intended to be used by delegatecalling to it from a DSProxy
+ */
+contract CozyInvestConvex is CozyInvestHelpers, ICozyInvest1, ICozyDivest1, ICozyReward {
+  using SafeERC20 for IERC20;
+
+  /// @notice The unprotected money market we borrow from / repay to
+  address public immutable moneyMarket;
+  /// @notice The protection market we borrow from / repay to
+  address public immutable protectionMarket;
+  /// @notice Underlying of the above money market and protection market
+  address public immutable underlying;
+  /// @notice Curve deposit zap
+  address public immutable curveDepositZap;
+  /// @notice Token index for the curve pool
+  int128 public immutable curveIndex;
+  /// @notice Curve LP token
+  address public immutable curveLpToken;
+  /// @notice Convex booster id
+  uint256 public immutable convexPoolId;
+  /// @notice Convex reward manager
+  address public immutable convexRewardManager;
+
+  /// @notice Curve metapool
+  bool public immutable longSigFormat;
```

Comment:
```suggestion
  /// @notice There are two different `add_liquidity` function signatures used by 
  // different Curve deposit zaps, so this tells the contract which one to use
  bool public immutable longSigFormat;
```

Reply:
we might also want to take a closer look at this contract for any other spots that could benefit from more explanations/comments for devs